tests/cibyl/e2e/test_jenkins.py

```diff
@@ -681,3 +682,206 @@ def test_filter_ip_show_last_build(self):
+        expected.add('Security group mechanism: native ovn', 5)
+        expected.add('DVR: False', 5)
+        expected.add('TLS everywhere: N/A', 5)
+        expected.add('Storage: ', 4)
+        expected.add('Cinder backend: nfs', 5)
+        expected.add('Ironic: ', 4)
+        expected.add('Ironic inspector: False', 5)
+        expected.add('Cleaning network: False', 5)
+        expected.add('Overcloud templates: N/A', 4)
+        expected.add('Testing information: ', 4)
+        expected.add('Test suites: ', 5)
+        expected.add('- neutron', 6)
+        expected.add('- octavia', 6)
+        expected.add('Setup: rpm', 5)
+        expected.add('Total jobs found in query: 1', 2)
+
+        self.assertIn(expected.build(), self.stdout)
+
+    def test_get_spec_verbose_job_2(self):
+        """Check the spec jobs is correctly retrieved and all fields are
+        printed with verbose output."""
+        sys.argv = [
+            'cibyl',
+            '--config', 'tests/cibyl/e2e/data/configs/jenkins.yaml',
+            '-p', 'openstack', '-vv',
+            '-f', 'text', 'spec', 'test_2'
+        ]
+
+        main()
+
+        expected = IndentedTextBuilder()
+        expected.add('Job: test_2', 2)
+        expected.add('URL: http://localhost:8080/job/test_2/', 3)
+        expected.add('Openstack deployment: ', 3)
+        expected.add('Release: 16.2', 4)
+        expected.add('Infra type: ovb', 4)
+        expected.add('Topology: N/A', 4)
+        expected.add('Network: ', 4)
+        expected.add('IP version: 6', 5)
+        expected.add('Network backend: N/A', 5)
+        expected.add('ML2 driver: ovs', 5)
+        expected.add('Security group mechanism: iptables hybrid', 5)
+        expected.add('DVR: N/A', 5)
+        expected.add('TLS everywhere: True', 5)
+        expected.add('Storage: ', 4)
+        expected.add('Cinder backend: swift', 5)
+        expected.add('Ironic: ', 4)
+        expected.add('Ironic inspector: True', 5)
+        expected.add('Cleaning network: True', 5)
+        expected.add('Overcloud templates: N/A', 4)
+        expected.add('Testing information: N/A', 4)
+        expected.add('Total jobs found in query: 1', 2)
+
+        self.assertIn(expected.build(), self.stdout)
+
+    def test_get_spec_non_existing_job(self):
+        """Check the spec command fails if requesting a job that does not
+        exists."""
+        sys.argv = [
+            'cibyl',
+            '--config', 'tests/cibyl/e2e/data/configs/jenkins.yaml',
+            '-p', 'openstack',
+            '-f', 'text', 'spec', 'non_existing_job'
+        ]
+
+        main()
+        self.assertIn("No query performed", self.stdout)
+
+    def test_get_spec_multiple_jobs(self):
+        """Check the spec command fails if requesting the spec for many
+        jobs."""
+        sys.argv = [
+            'cibyl',
+            '--config', 'tests/cibyl/e2e/data/configs/jenkins.yaml',
+            '-p', 'openstack',
+            '-f', 'text', 'spec', 'test'
```

Comment:
In this case which are the multiple jobs?

Reply:
There are several jobs in the e2e testing environment that would match the pattern `test` (test_1, test_2, test_3, ...)

Reply:
That's true.